node.go

```diff
@@ -116,32 +115,30 @@ func (n *Node) EnableMDNSDiscovery(notifee discovery.Notifee, interval time.Dura
 }
 
 // NewNode creates a new ipfs basichost node with given identity
-func NewNode(listenAddr string, id peer.ID, priv ic.PrivKey) (node *Node, err error) {
-	var n Node
-	n.NetAddr, err = ma.NewMultiaddr(listenAddr)
+func NewNode(listenAddr string, agent *LibP2PAgent) (node *Node, err error) {
+
+	nodeID, _, err := agent.NodeID()
 	if err != nil {
 		return
 	}
 
-	ps := pstore.NewPeerstore()
-	pid, err := peer.IDFromPrivateKey(priv)
+	var n Node
+	n.NetAddr, err = ma.NewMultiaddr(listenAddr)
 	if err != nil {
 		return
 	}
 
-	if pid.String() != id.String() {
-		err = errors.New("NewNode: Id doesn't match key")
-		return
-	}
+	ps := pstore.NewPeerstore()
 
-	n.HashAddr = pid
-	ps.AddPrivKey(pid, priv)
-	ps.AddPubKey(pid, priv.GetPublic())
+	n.HashAddr = nodeID
+	priv := agent.PrivKey()
+	ps.AddPrivKey(nodeID, priv)
```

Comment:
h.id wasn't the public key, it's the hash of the public key.

name change because it really isn't based on IPFS, but based on libp2p addressing scheme.  This interface (the Agent) is really only a partial abstraction, because we are still using peer.ID (i.e. libp2p's notion of a node ID)